$DragonFly: src/gnu/usr.bin/cc34/cc_prep/patches/gcse.c.patch,v 1.2 2004/06/20 00:10:52 joerg Exp $

diff -u ../gcc.orig/gcse.c gcse.c
--- ../gcc.orig/gcse.c	2004-06-19 22:27:11.000000000 +0200
+++ gcse.c	2004-06-20 01:16:04.000000000 +0200
@@ -4178,7 +4178,7 @@
       /* Find an assignment that sets reg_used and is available
 	 at the start of the block.  */
       set = find_avail_set (regno, insn);
-      if (! set)
+      if (! set || SET_VOLATILE_P (set->expr))
 	continue;
 
       pat = set->expr;
